From b49ffd2b72e6d4d73a74752feeb54094adeff45d Mon Sep 17 00:00:00 2001
From: Rohit Rangwani <rrangwan@codeaurora.org>
Date: Wed, 14 Mar 2018 12:21:38 +0530
Subject: [PATCH 01/14] NFC: Add nfc data file context and rename property

sys.nfc.nq property prefix renamed to vendor.qti.nfc
as required for Android P upgrade.

Add nfc_data_file and nfc_vendor_data_file file contexts
to allow access to /data/nfc and /data/vendor/nfc files
repectively.

Change-Id: I0f26278a6527a4fd96e3362ae979bf794d08bfa5
---
 common/file.te           | 3 +++
 common/file_contexts     | 1 +
 common/hal_nfc.te        | 1 +
 common/property_contexts | 2 +-
 4 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/common/file.te b/common/file.te
index 9c083aa..ce30801 100644
--- a/common/file.te
+++ b/common/file.te
@@ -248,6 +248,9 @@ type persist_alarm_file, file_type;
 
 type persist_time_file, file_type;
 
+# nfc file type for data vendor access
+type nfc_vendor_data_file, file_type, data_file_type;
+
 # kgsl file type for sysfs access
 type sysfs_kgsl, sysfs_type, fs_type;
 type sysfs_kgsl_proc, sysfs_type, fs_type;
diff --git a/common/file_contexts b/common/file_contexts
index afbe22e..844cab5 100644
--- a/common/file_contexts
+++ b/common/file_contexts
@@ -511,6 +511,7 @@
 # data files
 #
 /data/connectivity(/.*)?                                            u:object_r:cnd_data_file:s0
+/data/vendor/nfc(/.*)?                                              u:object_r:nfc_vendor_data_file:s0
 /data/vendor/connectivity(/.*)?                                     u:object_r:cnd_data_file:s0
 /data/data_test(/.*)?                                               u:object_r:data_test_data_file:s0
 /data/diag_log(/.*)?                                                u:object_r:diag_data_file:s0
diff --git a/common/hal_nfc.te b/common/hal_nfc.te
index 801a19a..e4dcada 100644
--- a/common/hal_nfc.te
+++ b/common/hal_nfc.te
@@ -31,3 +31,4 @@ set_prop(hal_nfc, nfc_nq_prop)
 #Allow access to firmware
 allow hal_nfc firmware_file:dir r_dir_perms;
 allow hal_nfc firmware_file:file r_file_perms;
+allow hal_nfc nfc_vendor_data_file:file rw_file_perms;
diff --git a/common/property_contexts b/common/property_contexts
index 4b89c63..073395e 100644
--- a/common/property_contexts
+++ b/common/property_contexts
@@ -106,7 +106,7 @@ sys.boot_mode              u:object_r:boot_mode_prop:s0
 # GPU
 ro.gpu.available_frequencies u:object_r:freq_prop:s0
 # NFC
-sys.nfc.nq.                u:object_r:nfc_nq_prop:s0
+vendor.qti.nfc.            u:object_r:nfc_nq_prop:s0
 ctl.ppd                    u:object_r:ppd_prop:s0
 qemu.gles                  u:object_r:qemu_gles_prop:s0
 # LKCore start
-- 
2.17.1

